tests/storage/test_registration.py

```diff
@@ -116,3 +117,29 @@ def test_is_support_user(self):
         )
         res = yield self.store.is_support_user(SUPPORT_USER)
         self.assertTrue(res)
+
+    @defer.inlineCallbacks
+    def test_3pid_inhibit_invalid_validation_session_error(self):
+        """Tests that enabling the configuration option to inhibit 3PID errors on
+        /requestToken also inhibits validation errors caused by an unknown session ID.
+        """
+
+        # Check that, with the config setting set to false (the default value), a
+        # validation error is caused by the unknown session ID.
+        try:
+            yield self.store.validate_threepid_session(
```

Comment:
Could you update these to `await`s now that the clokep has ploughed through the codebase? :P

Reply:
I don't think you can use `async` functions in tests?

These calls need an `defer.ensureDeferred` around them though.

Reply:
> These calls need an defer.ensureDeferred around them though.

Oops, yes this is what I meant.

Reply:
`store.validate_threepid_session` isn't async though.